core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -136,6 +136,11 @@
     private static final String UPDATE_CENTER_URL = System.getProperty(UpdateCenter.class.getName()+".updateCenterUrl","http://updates.jenkins-ci.org/");
 
     /**
+     * Read timeout when downloading plugins, defaults to 1 minute
+     */
+    private static final int PLUGIN_DOWNLOAD_READ_TIMEOUT = Integer.getInteger(UpdateCenter.class.getName()+".pluginDownloadReadTimeoutSeconds", 60) * 1000;
```

Comment:
Any new system property should be documented [here](https://wiki.jenkins-ci.org/display/JENKINS/Features+controlled+by+system+properties) - after release of course.